plugin/core/types.py

```diff
@@ -506,6 +506,8 @@ def __init__(self,
                  command: Optional[List[str]] = None,
                  binary_args: Optional[List[str]] = None,  # DEPRECATED
                  tcp_port: Optional[int] = None,
+                 auto_complete_selector: Optional[str] = None,
+                 allow_completion_triggers_from_server: bool = True,
```

Comment:
Since we have 2 auto-complete-related options already, maybe we could group them under the same object?

Like:

```
"auto_complete": {
  "selector": "punctuation",
  "ignore_server_triggers": True
}
```

Just a suggestion so that things are a little less messy.

Reply:
Also, I find the "default false" to be a better approach since it's more clear when key being missing == False.

Reply:
I agree with using `ignore_server_triggers`. But I don't like having another object as config item. I'm actually trying to flatten the config because I feel that users have trouble understanding the mental model. Which is understandable, just look at our sublime-package.json. I hope to be able to write

```
"disabled_capabilities": ["completionProvider.triggerCharacters"]
```

some time in the future (but not in this PR)